engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -908,7 +908,7 @@ private void execute() throws YarnException, IOException
           LOG.info("Releasing {} as resource with priority {} was already assigned", allocatedContainer.getId(), allocatedContainer.getPriority());
           releasedContainers.add(allocatedContainer.getId());
           numReleasedContainers++;
-          numRequestedContainers--;
+          numRequestedContainers++;    // undo the decrement above for this allocated container
```

Comment:
@vrozov I see that this is merged but just to complete the discussion: whether this is a corner case or not can be subjective so I don't want to dwell on it. All I am saying is that the situation/bug you described was always there and this JIRA or PR wasn't trying to address it. We will need a different JIRA and probably a refactor along the lines suggested in APEXCORE-771 will address this.